prac_10/recursion_pyramid.py

```diff
@@ -0,0 +1,27 @@
+"""
+CP1404/CP5632 Practical
+Recursion Pyramid
+"""
+
+
+# # LOOP VERSION
+def block_step(rows):
+    total = 0
+    for i in range(rows):
+        total += rows
+        rows -= 1
```

Comment:
`rows` should not be changed inside the loop. You should be incrementing total by i, not rows.